habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -1326,3 +1330,242 @@ def obj_next_to(
+        matching_recs = [
+            rec
+            for u_name, rec in candidate_receptacles.items()
+            if support_object.handle in u_name
+        ]
+        if support_object.object_id != support_surface_id:
+            # support object is a link
+            link_index = support_object.link_object_ids[support_surface_id]
+            # further cull the list to this link's recs
+            matching_recs = [
+                rec for rec in matching_recs if rec.parent_link == link_index
+            ]
+        if len(matching_recs) == 0:
+            # there are no Receptacles for this support surface
+            try_floor = True
+        else:
+            # select a Receptacle which most likely contains the point
+            dist_to_recs = [
+                rec.dist_to_rec(sim, obj_bottom_point) for rec in matching_recs
+            ]
+
+            # define an epsilon distance similarity threshold for determining two receptacles are equi-distant (e.g. overlapping)
+            same_dist_eps = 0.01
+            min_indices = []
+            min_dist: float = None
+            for ix, dist in enumerate(dist_to_recs):
+                if dist < max_dist_to_rec:
+                    if min_dist is None or dist < (min_dist - same_dist_eps):
+                        min_dist = dist
+                        min_indices = [ix]
+                    elif abs(dist - min_dist) < same_dist_eps:
+                        # this rec is nearly equidistant to the current best match
+                        min_indices.append(ix)
+
+            if len(min_indices) > 0:
+                return (
+                    [matching_recs[ix].unique_name for ix in min_indices],
+                    1.0 - (min_dist / max_dist_to_rec),
+                    "successful match",
+                )
+            else:
+                info_text = "Point is too far from a valid Receptacle on the support surface."
+
+    # check if the point is navigable and if so, try matching it to a region
+    if try_floor:
+        snap_point = sim.pathfinder.snap_point(obj_bottom_point, island_index)
+        snap_point[1] = obj_bottom_point[1]
+        if (obj_bottom_point - snap_point).length() < 0.1:
+            # this point is on the floor and should be mapped to a region
+            point_regions = sim.semantic_scene.get_weighted_regions_for_point(
+                obj_bottom_point
+            )
+            if len(point_regions) > 0:
+                # found matching regions, pick the primary (most precise) one
+                region_name = sim.semantic_scene.regions[
+                    point_regions[0][0]
+                ].id
+            else:
+                # point is not matched to a region
+                region_name = "unknown_region"
+            return [f"floor,{region_name}"], 1.0, "successful match"
+        else:
+            info_text = (
+                "Point does not match any Receptacle and is not navigable."
```

Comment:
It may be worthwhile to move this check outside of this function.

This check includes a very common failure, where users place objects in non-default receptacles.
This condition is specific to the ongoing set of experiments, therefore should be toggled via config.

Furthermore, this error requires additional UI feedback (e.g. highlighting).